gpio/imx6ull-gpio/imx6ull-gpio.c

```diff
@@ -255,10 +256,49 @@ void thread(void *arg)
 }
 
 
-int main(void)
+static int parse_args(int argc, char *argv[])
+{
+	int opt;
+	unsigned long prio;
+	char *endptr;
+
+	while ((opt = getopt(argc, argv, "p:")) != -1) {
+		switch (opt) {
+			case 'p':
+				prio = strtoul(optarg, &endptr, 0);
+				if (endptr == optarg) {
+					printf("%s: incorrect priority value (%s)\n", argv[0], optarg);
+					return -1;
+				}
+
+				if (prio > 7) {
+					printf("%s: incorrect priority value (%lu). It must be in [0,7] range\n",
+						argv[0], prio);
+					return -1;
+				}
+
+				priority(prio);
+				break;
+
+			default:
+				printf("gpio driver for imx6ull. Usage:\n");
+				printf("%s -p priority\n", argv[0]);
+				printf("\t-p priority\t\tSelect priority for the gpio driver\n");
+				return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+int main(int argc, char *argv[])
 {
 	oid_t root;
 
+	if (parse_args(argc, argv) < 0)
+		return -1;
```

Comment:
don't return negative values from main(). Actually only 0-127 values should be used for user return codes (`wait`/`waitpid` returns only lower 8 bytes of child process return code and return code = 128 + SIGNO usually means "killed by signal SIGNO")